stingray/powerspectrum.py

```diff
@@ -399,3 +399,77 @@ def _make_segment_spectrum(self, lc, segment_size):
             nphots_all.append(np.sum(lc_seg.counts))
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac", gti=None):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Attributes
+        ----------
+        dyn_ps : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        self.gti = gti
+        self.power_all, _ = AveragedPowerspectrum._make_segment_spectrum(
```

Comment:
(this might be to much nitpicking from my part, but..)

I find the name `power_all` confusing. It is a collection of powerspectra (not only powers) can you thing of a more descriptive name (but still short)? The best I could come up with was `pss` or `ps_collection` or maybe better `ps_all` or `all_ps`.

I think it is not unreasonable to have a variable which stores all the powers of the individual powerspectra, and that I would call `power_all`   (e.g `self.power_all = np.array([p.power for p in self.ps_all])`). This would allow to easily plot the DynamicPowerspectrum by calling `plt.imshow(dynps.power_all)` and use the matryx `dyn_ps` as indexes.